client/lib/cgutil/cgutil_linux.go

```diff
@@ -0,0 +1,140 @@
+package cgutil
+
+import (
+	"os"
+	"path/filepath"
+
+	cgroupFs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	DefaultCgroupParent    = "/nomad"
+	SharedCpusetCgroupName = "shared"
+)
+
+// InitCpusetParent checks that the cgroup parent and expected child cgroups have been created
+// If the cgroup parent is set to /nomad then this will ensure that the /nomad/shared
+// cgroup is initialized. The /nomad/reserved cgroup will be lazily created when a workload
+// with reserved cores is created
```

Comment:
```suggestion
// InitCpusetParent checks that the cgroup parent and expected child cgroups have been created.
// If the cgroup parent is set to /nomad then this will ensure that the /nomad/shared
// cgroup is initialized. The /nomad/reserved cgroup will be lazily created when a workload
// with reserved cores is created.
```